File: Models/Vehicle.cs
using System.Text.Json.Serialization;
using Microcharts;
using System.Globalization; // NOVO: IMPORT NECESSÁRIO

namespace controle_ja_mobile.Models
{
    public class Vehicle
    {
        [JsonPropertyName("id")]
        public Guid Id { get; set; }

        [JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;

        [JsonPropertyName("brand")]
        public string Brand { get; set; } = string.Empty;

        [JsonPropertyName("model")]
        public string Model { get; set; } = string.Empty;

        [JsonPropertyName("year")]
        public int Year { get; set; }

        [JsonPropertyName("plate")]
        public string Plate { get; set; } = string.Empty;

        [JsonPropertyName("currentOdometer")]
        public decimal CurrentOdometer { get; set; }

        // Campos de Inteligência
        [JsonPropertyName("avgGasoline")]
        public double? AvgGasoline { get; set; }

        [JsonPropertyName("avgEthanol")]
        public double? AvgEthanol { get; set; }

        [JsonIgnore]
        public string FullDescription => $"{Brand} {Model} - {Year}";

        // FIX: Força formatação brasileira de inteiro (Ex: 179.327 km, not 179,327 km)
        [JsonIgnore]
        public string FormattedOdometer => CurrentOdometer.ToString("N0", new CultureInfo("pt-BR")) + " km";

        [JsonIgnore]
        public string ConsumptionInfo
        {
            get
            {
                string info = "";
                if (AvgGasoline > 0) info += $"Gas: {AvgGasoline:F1} km/L\n";
                if (AvgEthanol > 0) info += $"Eta: {AvgEthanol:F1} km/L";
                return string.IsNullOrWhiteSpace(info) ? "Sem dados" : info.Trim();
            }
        }

        [JsonIgnore]
        public string MonthlyCost { get; set; } = "R$ 0,00";

        [JsonIgnore]
        public Chart ExpensesChart { get; set; }
    }
}

File: lib/models/vehicle.dart
import 'enums.dart';

class Vehicle {
  final String id;
  final String name;
  final String brand;
  final String model;
  final int year;
  final String plate;
  final double currentOdometer;
  final double? avgGasoline;
  final double? avgEthanol;

  const Vehicle({
    required this.id,
    this.name = '',
    this.brand = '',
    this.model = '',
    this.year = 0,
    this.plate = '',
    this.currentOdometer = 0,
    this.avgGasoline,
    this.avgEthanol,
  });

  String get fullDescription => '$brand $model - $year';

  Vehicle copyWith({
    String? id,
    String? name,
    String? brand,
    String? model,
    int? year,
    String? plate,
    double? currentOdometer,
    double? avgGasoline,
    double? avgEthanol,
  }) {
    return Vehicle(
      id: id ?? this.id,
      name: name ?? this.name,
      brand: brand ?? this.brand,
      model: model ?? this.model,
      year: year ?? this.year,
      plate: plate ?? this.plate,
      currentOdometer: currentOdometer ?? this.currentOdometer,
      avgGasoline: avgGasoline ?? this.avgGasoline,
      avgEthanol: avgEthanol ?? this.avgEthanol,
    );
  }

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] as String? ?? '',
      name: json['name'] as String? ?? '',
      brand: json['brand'] as String? ?? '',
      model: json['model'] as String? ?? '',
      year: json['year'] as int? ?? 0,
      plate: json['plate'] as String? ?? '',
      currentOdometer:
          (json['currentOdometer'] as num?)?.toDouble() ?? 0,
      avgGasoline: (json['avgGasoline'] as num?)?.toDouble(),
      avgEthanol: (json['avgEthanol'] as num?)?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'brand': brand,
      'model': model,
      'year': year,
      'plate': plate,
      'currentOdometer': currentOdometer,
      'avgGasoline': avgGasoline,
      'avgEthanol': avgEthanol,
    };
  }

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is Vehicle && runtimeType == other.runtimeType && id == other.id;

  @override
  int get hashCode => id.hashCode;

  @override
  String toString() =>
      'Vehicle(id: $id, name: $name, brand: $brand, model: $model, '
      'year: $year, plate: $plate, currentOdometer: $currentOdometer)';
}
